gcc/rust/typecheck/rust-hir-type-check-item.cc

```diff
@@ -518,6 +518,20 @@ TypeCheckItem::visit (HIR::Module &module)
 void
 TypeCheckItem::visit (HIR::Trait &trait)
 {
+  if (trait.has_type_param_bounds ())
+    {
+      for (auto &tp_bound : trait.get_type_param_bounds ())
+        {
+          HIR::TraitBound &tb
+              = static_cast<HIR::TraitBound &> (*tp_bound.get ());
+          if (tb.get_polarity () == BoundPolarity::AntiBound)
```

Comment:
this is good, but the static cast might fail if the trait bound is a lifetime.

you can switch on `tp_bound.get_bound_type()` and check if it is a lifetime or a trait, and proceed to cast safely then. so something like:
```suggestion
          if (tp_bound.get_bound_type() == LIFETIME) {
          HIR::TraitBound &tb
              = static_cast<HIR::TraitBound &> (*tp_bound.get ());
          if (tb.get_polarity () == BoundPolarity::AntiBound)
          /* snip */
          }
```

etc. does that make sense?

Reply:
@CohenArthur, thank you for the review. Do  you have an example of the when the static cast might fail?

This program seems to compile fine with the code as is:

```rust
#[lang = "sized"]
pub trait Sized {}
trait Trait<'a>: 'a {}
```

Reply:
ah, my bad - the static cast won't fail, but the value will be undefined. so we still need to do the check.